### ISSUE TITLE
Destructuring an object with not specified fields on type not caught

### ISSUE THREAD

**Issue by HoneySpace** [microsoft/TypeScript#60714]
### 🔎 Search Terms

Destructuring object, not specified fields, not caught

### 🕗 Version & Regression Information

This is the behavior in every version I tried, and I reviewed the FAQ for entries about Common "Bugs" That Aren't Bugs

### ⏯ Playground Link

https://www.typescriptlang.org/play/?#code/C4TwDgpgBAglC8UDeBDAXAOwK4FsBGEATgDRR6a4GEC+AULQMYD2GAzsFCgsugIykM0AMxQAbVhGpRGLdmTQAFFIWABLMQB4YAPm6o0AJjrM2HQUpXrRW3YiQA6RymOyOAE0XK1mnXsf2UYkEAZjogA

### 💻 Code

```ts
type A = {a:number, b:number}

const a = {a:1, c:false} 
const b:Partial<A> = {a:2}
const c:Partial<A> = {...a} // doesn't catch field "c" being not specified on type A
const d:Partial<A> = {...a, c:3} // prevents specifying field not from type A
```

### 🙁 Actual behavior

Object that partially matches type structure was allowed to be destructured into object with defined type, despite having a field that not specified on mentioned type.
Meanwhile when defining not specified field directly on object Typescript will throw an error "Object literal may only specify known properties"

### 🙂 Expected behavior

Since object has a not specified property from type, it should throw an error, same as defining a not specified field directly on object
Both destructuring an defining  not specified property lead to object have property that doesn't match type structure.

Not catching error when destructuring can easily lead to incorrect behavior,  as an example, using function with a lot of parameters that meant to be passed to next function, and describing new property that doesn't exist on type of next function parameters leads function not accounting for this property or even processing it when it shouldn't.

**Comment by MartinJohns** [comment]
Duplicate of #58835 and many others.

**Comment by jcalz** [comment]
Terminology note: this is [spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), not [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#rest_property).
